src/router.js

```diff
@@ -1,28 +1,47 @@
 import Vue from 'vue'
 import Router from 'vue-router'
 
+import localStorage from '@/services/localStorage'
+
 Vue.use(Router)
 
-export default new Router({
+const router = new Router({
   routes: [
     {
       path: '/sign-up',
       name: 'signUp',
-      component: () => import('./views/SignUp.vue')
+      component: () => import('./views/SignUp.vue'),
+      meta: { requiresAuth: false }
     },
     {
       path: '/login',
       name: 'login',
-      component: () => import('./views/Login.vue')
+      component: () => import('./views/Login.vue'),
+      meta: { requiresAuth: false }
     },
     {
       path: '/dashboard',
       name: 'dashboard',
-      component: () => import('./views/Dashboard.vue')
+      component: () => import('./views/Dashboard.vue'),
+      meta: { requiresAuth: true }
     },
     {
       path: '/',
       redirect: '/login'
     }
   ]
 })
+
+router.beforeEach((to, from, next) => {
+  const isLogged = localStorage.getSessionData()
+  const isRootPath = to.path === '/'
+  debugger
+  if ((to.meta.requiresAuth || isRootPath) && !isLogged) {
+    next({ path: '/login' })
+  } else if ((!to.meta.requiresAuth || isRootPath) && isLogged) {
+    next({ path: '/dashboard' })
```

Comment:
you can now rename the `/dasboard` path to `/` and remove this conditional branch